Source/EasyNetQ/AutoSubscribe/DefaultAutoSubscriberMessageDispatcher.cs

```diff
@@ -1,24 +1,20 @@
-using EasyNetQ.DI;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace EasyNetQ.AutoSubscribe;
 
 public class DefaultAutoSubscriberMessageDispatcher : IAutoSubscriberMessageDispatcher
 {
-    private readonly IServiceResolver resolver;
+    private readonly IServiceProvider resolver;
 
-    public DefaultAutoSubscriberMessageDispatcher(IServiceResolver resolver) => this.resolver = resolver;
-
-    public DefaultAutoSubscriberMessageDispatcher() : this(new ActivatorBasedResolver())
```

Comment:
Why this constructor has been removed?

Reply:
https://github.com/EasyNetQ/EasyNetQ/wiki/Auto-Subscriber

@Atallah0 Please read about this feature. I am not sure how broadly it is used, anyway we need to discuss and decide what to do with it.

Reply:
I think that if we make the use of a DI container required everywhere, I also don't think it makes much sense to have an ActivatorBasedResolver there. Consumers should be resolved from the container.

Reply:
It makes sense. 

However, we need to rework Autosubscriber then. It accepts `Type[] consumerTypes` and it doesn't make any sense, because all implementations of IConsume and IConsumeAsync interfaces can be injected directly to Autosubscriber via its constructor.

Also, we need a dedicated documentation for v8 which demonstrates how to migrate and how to use this new Autosubscriber.

PS @zidad  mentioned that would like to have BackgroundService to start EasyNetQ and it seems that Autosubscriber should be part of it.

Reply:
as we also discussed in slack, I would like to keep the AutoSubscriber isolated from the core, and maybe even a separate assembly, because it's dependent on reflection which (I think) makes it impossible to use AOT at some point